### DISCUSSION TITLE
Are touch gestures supported in this library?

### DISCUSSION THREAD

**Discussion by liorp** [doronz88/pymobiledevice3#358] [Q&A]
Hi!
As the title says, is multi touch gestures work in this library? Or is there another library which implements interactions with a physical device?

**Comment by doronz88** [comment]
no, you'll need to jailbreak the device to use. apple don't provide an api for it. do you need this for a jailbroken device? over xcuitest?

**Reply by liorp** [reply]
Thanks for the swift reply! I’d need this to work on a non jailbroken device. I’ve seen that appium implements something similar over XCUITest, and I wonder if it can be done via more “low level” interface w.r.t Python, or whether I should just stick with appium.

**Comment by doronz88** [comment]
unfortunately you cannot. xcuitest (appium) is the solution apple provided for it
